src/cr/cube/matrix.py

```diff
@@ -224,20 +317,29 @@ def _inserted_rows(self):
     @lazyproperty
     def _row_order(self):
         """ -> 1D ndarray of int row idx specifying order of unordered-array rows."""
-        # ---Specifying int type prevents failure when there are zero rows---
+        # --- specifying int type prevents failure when there are zero rows ---
         return np.array(self._rows_dimension.display_order, dtype=int)
```

Comment:
that’s … an interesting side effect of dtype. why is this the case? do we ever go as far as calling row_order on an empty table?

Reply:
Adding the `dtype=int` (and the comment) actually came as a fix to a Sentry error that would come up in rashes maybe a year and a half ago. We don't have conditional logic that prevents a table with zero rows or zero columns from being processed, and `._row_order` is a basic attribute that is accessed unconditionally in the course of all `_Slice` computations, so yeah, we need it to work for a table with no valid rows.

I expect the error came up in the course of tabbook exports where folks were selecting a bunch of variables without necessarily looking too closely at them.